File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.cpp
#include <iostream>
#include<vector>
#include<algorithm>
using namespace std;

int solution(vector<int> A, vector<int> B)
{
    int answer = 0;

    sort(A.begin(), A.end());
    sort(B.begin(), B.end());
    
    int n = A.size();
    for(int i = 0; i < n; i++)
    {
        answer += A[i] * B[n-i-1];
    }
    
    int temp = 0;
    for(int i = 0; i < n; i++)
    {
        temp += A[n-i-1] * B[i];
    }
    
    answer = min(answer, temp);
    
    return answer;
}

File: 프로그래머스/2/12941. 최솟값 만들기/최솟값 만들기.cs
using System;
using System.Linq;

public class Solution {
    public int solution(int[] A, int[] B) {
        int answer = 0;
        
        Array.Sort(A);
        Array.Sort(B);
        
        int size = A.Length;
        for(int i= 0 ; i < size; i++)
        {
            answer += A[i] * B[size - 1 - i];
        }
        
        return answer;
    }
}